avocado/plugins/resolvers.py

```diff
@@ -110,6 +110,25 @@ def resolve(reference):
                                find_avocado_tests)
 
 
+class RequirementsResolver:
```

Comment:
I find the location you chose for the `RequirementResolver` very confusing.  This file contains only *test reference* resolvers, and they are all based on `Resolver`, and their `resolve` method is very different with regards to input and output.

Also, the `RequirementResolver` here can not be said to be a plugin IMO.  It's something that the core suite/job/statemachine will use.  Maybe when we introduce a way to read and resolve requirements from other sources, and implement a plugin interface, then it'll make sense to have them under `avocado/plugins`.

Reply:
As the code became shorter than the previous version, I thought about concentrating the resolvers we have now on the same location. I forgot to change the file docstring to reflect that. I don't think it is a problem to add it to the `avocado.core`, it is just a matter of code organization. I'll try to find a better place for it inside the `core`.

Reply:
I agree with @clebergnu and I think we might have mentioned this before - even though we use the same word "resolver" (one reason for creating this confusion), the requirements resolver component is meant for a very different use. Perhaps this could be resolve later on if the test resolvers are named back "loaders" once the serial loader is deprecated.

Reply:
Okay, got it!

Reply:
> I agree with @clebergnu and I think we might have mentioned this before - even though we use the same word "resolver" (one reason for creating this confusion), the requirements resolver component is meant for a very different use. Perhaps this could be resolve later on if the test resolvers are named back "loaders" once the serial loader is deprecated.

I agree in the part that you agree with me :) and also I can also relate to some of the confusion.  But, we do have a prefix somehow distinguishing it from the "common" resolvers.  Also, I'd try to stay away from renaming the resolvers `loaders` because:

* It doesn't really load any tests, but simply describes how one can "reach" and eventually run them
* It may lead people into thinking that the old architecture is back (or never left)

Reply:
> I agree with the part that you agree with me :)

:smile: 

> and also I can also relate to some of the confusion.

My confusion was cleared below, I think the current approach is fine so long as the different resolvers and runners converge well enough.

>  Also, I'd try to stay away from renaming the resolvers `loaders` because:
> 
>     * It doesn't really load any tests, but simply describes how one can "reach" and eventually run them

Fair enough. I suggested this since your original motivation for the "resolver" name before was simply to distinguish it from loaders. This newer argument makes more sense than the previous one in fact.

>     * It may lead people into thinking that the old architecture is back (or never left)

This argument is not as valid though since we are not supposed to use nomenclature in code to distinguish versions and mingle version information together with implementation logic. We have changelogs, release tags, and what else not to express this information. So having expectation that class names imply version information is not good to begin with. But yes, the first argument above brings some fresh air to the choice of name.